classes/submission/maps/Schema.php

```diff
@@ -559,19 +562,21 @@ protected function mapByProperties(array $props, Submission $submission, bool|Co
     /**
      * Determine whether current user is able to change metadata
      */
-    protected function canChangeMetadata(Enumerable $stageAssignments, Submission $submission): bool
+    protected function canChangeMetadata(?Enumerable $stageAssignments): bool
     {
         $currentUser = Application::get()->getRequest()->getUser();
         $isAssigned = false;
         $canChangeMetadata = false;
 
         // Check if stage assignment is associated with the current user and edit metadata flag
-        foreach ($stageAssignments as $stageAssignment) {
-            if ($stageAssignment->userId === $currentUser->getId()) {
-                $isAssigned = true;
-                if ($stageAssignment->canChangeMetadata) {
-                    $canChangeMetadata = true;
-                    break;
+        if ($stageAssignments) {
```

Comment:
Unrelated to this change, but the comment on the line 588 and the following `if` sound suspicious: `// If user is ASSIGNED` then `if (!$isAssigned)`.

If it should be `if ($isAssigned)` then this line/update can also be simplified to `if (!$stageAssignments) { return false; }`

Reply:
Likely just the comment is wrong, the code logic looks correct to me